From c3e8b8aa4bb025e20dcf7da4c0281469c85824f5 Mon Sep 17 00:00:00 2001
From: Michael Simons <msimons@microsoft.com>
Date: Mon, 26 Jul 2021 15:59:32 +0000
Subject: [PATCH] Exclude PerformanceTests from source build

This addresses the following issue in the PerformanceTests:
src/PerformanceTests/Utilities/Common/CompilationHelper.cs(126,23): error CS0534: 'CompilationHelper.OptionsProvider' does not implement inherited abstract member 'AnalyzerConfigOptionsProvider.GlobalOptions.get'

This is a result of https://github.com/dotnet/roslyn/pull/44875 which introduced the new abstract member.

Since the PerformanceTests are not necessary for source-build they were excluded.
---
 src/PerformanceTests/Tests/PerformanceTests.csproj               | 1 +
 src/PerformanceTests/Utilities/CSharp/CSharpPerfUtilities.csproj | 1 +
 src/PerformanceTests/Utilities/Common/CommonPerfUtilities.csproj | 1 +
 .../Utilities/VisualBasic/VisualBasicPerfUtilities.csproj        | 1 +
 4 files changed, 4 insertions(+)

diff --git a/src/PerformanceTests/Tests/PerformanceTests.csproj b/src/PerformanceTests/Tests/PerformanceTests.csproj
index 5b32efdb2..c429c778a 100644
--- a/src/PerformanceTests/Tests/PerformanceTests.csproj
+++ b/src/PerformanceTests/Tests/PerformanceTests.csproj
@@ -14,6 +14,7 @@
     <Optimize>true</Optimize>
     <Configuration>Release</Configuration>
     <IsPackable>false</IsPackable>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="BenchmarkDotNet" Version="$(BenchmarkDotNetVersion)" />
diff --git a/src/PerformanceTests/Utilities/CSharp/CSharpPerfUtilities.csproj b/src/PerformanceTests/Utilities/CSharp/CSharpPerfUtilities.csproj
index 829353e98..67cacfa19 100644
--- a/src/PerformanceTests/Utilities/CSharp/CSharpPerfUtilities.csproj
+++ b/src/PerformanceTests/Utilities/CSharp/CSharpPerfUtilities.csproj
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <NonShipping>true</NonShipping>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
 
   <ItemGroup>
diff --git a/src/PerformanceTests/Utilities/Common/CommonPerfUtilities.csproj b/src/PerformanceTests/Utilities/Common/CommonPerfUtilities.csproj
index 6bce3d4f6..16f6f1c9c 100644
--- a/src/PerformanceTests/Utilities/Common/CommonPerfUtilities.csproj
+++ b/src/PerformanceTests/Utilities/Common/CommonPerfUtilities.csproj
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <NonShipping>true</NonShipping>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
 
   <ItemGroup>
diff --git a/src/PerformanceTests/Utilities/VisualBasic/VisualBasicPerfUtilities.csproj b/src/PerformanceTests/Utilities/VisualBasic/VisualBasicPerfUtilities.csproj
index ff85941e1..c88990d44 100644
--- a/src/PerformanceTests/Utilities/VisualBasic/VisualBasicPerfUtilities.csproj
+++ b/src/PerformanceTests/Utilities/VisualBasic/VisualBasicPerfUtilities.csproj
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <NonShipping>true</NonShipping>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
 
   <ItemGroup>
-- 
2.29.2

